azure/scope/identity.go

```diff
@@ -270,7 +271,7 @@ func createAzureIdentityWithBindings(ctx context.Context, azureIdentity *infrav1
 			Type:           azureIdentityType,
 			TenantID:       azureIdentity.Spec.TenantID,
 			ClientID:       azureIdentity.Spec.ClientID,
-			ClientPassword: azureIdentity.Spec.ClientSecret,
+			ClientPassword: ptr.Deref(azureIdentity.Spec.ClientSecret, corev1.SecretReference{}),
```

Comment:
Just curious, sorry for the fly by review. 
Do you know if this could create a race scenario where we CAPZ controller is trying to set an empty `corev1.SecretReference{}` and the Azure resource in the cluster is accepting it as a "make it empty" and marks `ClientPassword: nil` ?

Reply:
This field doesn't map to an Azure API field on an Azure resource AFAIK. This change is purely to handle `azureIdentity.Spec.ClientSecret` being a pointer and the AzureIdentity field not. I don't think this involves any change in behavior.

Reply:
Thanks!